File: server/src/main/scala/kpn/api/common/network/NetworkSummary.scala
package kpn.api.common.network

import kpn.api.common.RouteScope
import kpn.api.common.RouteType

case class NetworkSummary(
  name: Option[String],
  routeType: RouteType,
  routeScope: RouteScope,
  factCount: Long,
  nodeCount: Long,
  routeCount: Long,
  changeCount: Long
)

File: backend/api/src/main/java/kpn/api/common/network/NetworkSummary.java
package kpn.api.common.network;

import kpn.api.common.RouteScope;
import kpn.api.common.RouteType;

import java.util.Optional;

public record NetworkSummary(
  Optional<String> name,
  RouteType routeType,
  RouteScope routeScope,
  Long factCount,
  Long nodeCount,
  Long routeCount,
  Long changeCount
) {
}
